corehq/apps/sms/tasks.py

```diff
@@ -201,7 +195,7 @@ def handle_outgoing(msg):
     )
 
     if max_simultaneous_connections:
```

Comment:
should this be in a finally block in case `send_message_via_backend` raises an exception?

Reply:
send_message_via_backend is designed to not raise exceptions, but if something were to go wrong we'd just have to wait for the lock to time out before the slot becomes free again